Creational/Builder.cpp

```diff
@@ -0,0 +1,142 @@
+#include <iostream>
+
+/* Smartphone parts */
+class Camera {
+    public:
+        int megapixels;
+};
+
+class Memory {
+    public:
+        int size;
+};
+
+class Display {
+    public:
+        float size;
+};
+
+/* My product is Smartphone */
+class Smartphone {
+    public:
+        Camera*  camera;
+        Memory*  memory;
+        Display* display;
+
+        void specifications()
+        {
+            std::cout << "camera:  " << camera->megapixels << " megapixels" << std::endl;
+            std::cout << "memory:  " << memory->size << " GB" << std::endl;
+            std::cout << "display: " << display->size << " inches" << std::endl;
+        }
+};
+
+/* Builder class is responsible for making Smartphone parts */
+class Builder {
+    public:
+        virtual Camera* getCamera() = 0;
+        virtual Memory* getMemory() = 0;
+        virtual Display* getDisplay() = 0;
+};
+
+/* Director class is responsible for the whole process */
+class Director {
+    Builder* builder;
+
+    public:
+        void setBuilder(Builder* newBuilder) {
+            builder = newBuilder;
+        }
+
+        Smartphone* getSmartphone()
+        {
+            Smartphone* smartphone = new Smartphone();
+
+            smartphone->camera = builder->getCamera();
+            smartphone->memory = builder->getMemory();
+            smartphone->display = builder->getDisplay();
+
+            return smartphone;
+        }
+};
+
+/* Builder for Iphone 11pro Smartphone */
+class Iphone11proBuilder: public Builder {
```

Comment:
Your concrete builder should know the product which should be built (Iphone11pro*).
Your builder methods should add the specific part (Memory) of object and return it (Iphone11pro*) or nothing (void).
Usually, the builder class has GetResult method which returns the current state of the object (Iphone11pro*).